tests/integration_test/ota_pal/aws_test_ota_pal.c

```diff
@@ -0,0 +1,1137 @@
+        blocksWritten = otaPal_WriteBlock( &xOtaFile,
+                                           0,
+                                           ucDummyData,
+                                           sizeof( ucDummyData ) );
+        TEST_ASSERT_EQUAL( sizeof( ucDummyData ), blocksWritten );
+
+        /* Fill out an incorrect signature. */
+        xOtaFile.pSignature = &xSig;
+        xOtaFile.pSignature->size = ucInvalidSignatureLength;
+        memcpy( xOtaFile.pSignature->data, ucInvalidSignature, ucInvalidSignatureLength );
+        xOtaFile.pCertFilepath = ( uint8_t * ) otatestpalCERTIFICATE_FILE;
+
+        /* Try to close the file. */
+        xOtaStatus = otaPal_CloseFile( &xOtaFile );
+
+        if( ( OtaPalBadSignerCert != OTA_PAL_MAIN_ERR( xOtaStatus ) ) &&
+            ( OtaPalSignatureCheckFailed != OTA_PAL_MAIN_ERR( xOtaStatus ) ) &&
+            ( OtaPalFileClose != OTA_PAL_MAIN_ERR( xOtaStatus ) ) )
+        {
+            TEST_ASSERT_TRUE( 0 );
+        }
+    }
+}
+
+/**
+ * @brief Call otaPal_CloseFile with an invalid signature in the file context.
+ * The close is called when no blocks have been written to the file.
+ * Verify the correct OTA Agent level error code is returned from otaPal_CloseFile.
+ */
+TEST( Full_OTA_PAL, otaPal_CloseFile_InvalidSignatureNoBlockWritten )
+{
+    OtaPalStatus_t xOtaStatus;
+    Sig256_t xSig = { 0 };
+
+    /* Create a local file using the PAL. */
+    xOtaFile.pFilePath = ( uint8_t * ) otatestpalFIRMWARE_FILE;
+    xOtaStatus = otaPal_CreateFileForRx( &xOtaFile );
+    TEST_ASSERT_EQUAL( OtaPalSuccess, OTA_PAL_MAIN_ERR( xOtaStatus ) );
+
+    /* Fill out an incorrect signature. */
+    xOtaFile.pSignature = &xSig;
+    xOtaFile.pSignature->size = ucInvalidSignatureLength;
+    memcpy( xOtaFile.pSignature->data, ucInvalidSignature, ucInvalidSignatureLength );
+    xOtaFile.pCertFilepath = ( uint8_t * ) otatestpalCERTIFICATE_FILE;
+
+    /* We still want to close the file if the test fails somewhere here. */
+    if( TEST_PROTECT() )
+    {
+        /* Try to close the file. */
+        xOtaStatus = otaPal_CloseFile( &xOtaFile );
+
+        if( ( OtaPalBadSignerCert != OTA_PAL_MAIN_ERR( xOtaStatus ) ) &&
+            ( OtaPalSignatureCheckFailed != OTA_PAL_MAIN_ERR( xOtaStatus ) ) &&
+            ( OtaPalFileClose != OTA_PAL_MAIN_ERR( xOtaStatus ) ) )
+        {
+            TEST_ASSERT_TRUE( 0 );
+        }
+    }
+}
+
+/**
+ * @brief Call otaPal_CloseFile with a signature verification certificate path does
+ * not exist in the system. Verify the correct OTA Agent level error code is returned
+ * from otaPal_CloseFile.
+ *
+ * @note This test is only valid if your device uses a file system in your non-volatile memory.
+ * Some devices may revert to using aws_codesigner_certificate.h if a file is not found, but
+ * that option is not being enforced.
+ */
+TEST( Full_OTA_PAL, otaPal_CloseFile_NonexistingCodeSignerCertificate )
+{
+    #if ( otatestpalUSE_FILE_SYSTEM == 1 )
```

Comment:
This was a pre-existing test that I did not write, so I can only speculate.

My understanding is that this test is only valid if a device has a file system. This could be a normal if statement, but I think this makes it more clear that this will never run if the device does not support a file system.